### PR TITLE
WGL-8: Build Automatic Release Pipeline

### PR DESCRIPTION
**Pull request by cyrusrohani**
Added workflow publish.yml file that performs these actions at each push:
 - Check out
 - Increment version in package.json
 - Push changed package.json
 - Create tag based on json version
 - Setup nodejs
 - npm publish
 - Create release

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: "WebGL SDK Publish"
+
+on:
+    push:
+        branches:
+            - main
```
Comment (JoeLangLeiaInc) [inline .github/workflows/publish.yml]: this should be updated to branch: dev

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: "WebGL SDK Publish"
+
+on:
+    push:
```
Comment (JoeLangLeiaInc) [inline .github/workflows/publish.yml]: more of an open question: should this be every push or every release?

Using Github tags and releases:
https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository

We can sync the npm releases to our github release tags without needing a new tag for every push.

Big picture, do we want to release with every push or plan releases independently?

**Review by JoeLangLeiaInc — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (cyrusrohani) [reply .github/workflows/publish.yml]: Per your other question regarding push every release, this is now changed to trigger per release.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (cyrusrohani) [reply .github/workflows/publish.yml]: I made the required changes to trigger the publishing on release only. That will be more convenient for release planning.

**Comment by cyrusrohani** [comment]
Alright, the latest changes are made. I had problems with npm publish complaining about the package not existing in the registry, which I though was an authentication issue but was because the package version wasn't being incremented. I assume this is something that will occur manually during release, but if not, I can update the script to do it automatically.

**Review by JoeLangLeiaInc — APPROVED** [review]
Looks good to me!

**Pull request merged**